src/Microsoft.AspNetCore.Authentication.OpenIdConnect/Events/OpenIdConnectEvents.cs

```diff
@@ -39,17 +29,22 @@ public class OpenIdConnectEvents : RemoteAuthenticationEvents, IOpenIdConnectEve
         /// <summary>
         /// Invoked before redirecting to the identity provider to authenticate.
         /// </summary>
-        public Func<RedirectContext, Task> OnRedirectToAuthenticationEndpoint { get; set; } = context => Task.FromResult(0);
+        public Func<RedirectContext, Task> OnRedirectToIdentityProvider { get; set; } = context => Task.FromResult(0);
 
         /// <summary>
         /// Invoked before redirecting to the identity provider to sign out.
         /// </summary>
-        public Func<RedirectContext, Task> OnRedirectToEndSessionEndpoint { get; set; } = context => Task.FromResult(0);
+        public Func<RedirectContext, Task> OnRedirectToIdentityProviderForSignOut { get; set; } = context => Task.FromResult(0);
 
         /// <summary>
         /// Invoked after "authorization code" is redeemed for tokens at the token endpoint.
         /// </summary>
-        public Func<TokenResponseReceivedContext, Task> OnTokenResponseReceived { get; set; } = context => Task.FromResult(0);
+        public Func<TokenResponseMessageReceivedContext, Task> OnTokenResponseMessageReceived { get; set; } = context => Task.FromResult(0);
+
+        /// <summary>
+        /// Invoked when an IdToken has been validated and produced an AuthenticationTicket.
+        /// </summary>
+        public Func<TokenValidatedContext, Task> OnTokenValidated { get; set; } = context => Task.FromResult(0);
```

Comment:
FWIW, I'm not a huge fan of the new name, which will sound too restrictive for most people: the validation routine doesn't just validate the `id_token`, but also the access token, the code and the nonce.